filters/H5Zbitshuffle/src/H5Zbitshuffle.jl

```diff
@@ -0,0 +1,284 @@
+                end
+                nbytes_out = nbytes_uncomp
+                
+            else  #shuffle and compress
+                
+                ccall((:bshuf_write_uint64_BE,libbitshuffle),Cvoid,(Ptr{Cvoid},Cuint),out_buf,nbytes_uncomp)
+                ccall((:bshuf_write_uint32_BE,libbitshuffle),Cvoid,(Ptr{Cvoid},Cuint),out_buf+8,block_size*elem_size)
+                
+                if compress_flag == BSHUF_H5_COMPRESS_LZ4
+                    err = ccall((:bshuf_compress_lz4,libbitshuffle),Cint,
+                                (Ptr{Cvoid},Ptr{Cvoid},Cuint,Cuint,Cuint),
+                                in_buf,out_buf+12,size,elem_size,block_size)
+                else
+                    err = ccall((:bshuf_compress_zstd,libbitshuffle),Cint,
+                                (Ptr{Cvoid},Ptr{Cvoid},Cuint,Cuint,Cuint),
+                                in_buf,out_buf+12,size,elem_size,block_size)
+                end
+                
+                nbytes_out = err + 12
+            end
+        else # just the shuffle thanks
+            
+            if flags & H5Z_FLAG_REVERSE != 0
+                err = ccall((:bshuf_bitunshuffle,libbitshuffle),Cint,
+                            (Ptr{Cvoid},Ptr{Cvoid},Cuint,Cuint,Cuint),
+                            in_buf,out_buf,size,elem_size,block_size)
+            else
+                err = ccall((:bshuf_bitshuffle,libbitshuffle),Cint,
+                            (Ptr{Cvoid},Ptr{Cvoid},Cuint,Cuint,Cuint),
+                            in_buf,out_buf,size,elem_size,block_size)
+            end
+            
+            nbytes_out = nbytes
+        end
+        
+        # And wrap it up
+
+        if err < 0
+            error("h5plugin_bitshuffle: Error in bitshuffle with code $err")
+        end
+        
+        Libc.free(unsafe_load(buf))
+        unsafe_store!(buf,out_buf)
+        unsafe_store!(buf_size,Csize_t(buf_size_out))
+        out_buf = C_NULL
+        
+    catch e
+
+        # On failure, return 0 and change no arguments
+        
+        nbytes_out = Csize_t(0)
+        @error "Non-fatal H5 bitshuffle plugin error: " e
+        display(stacktrace(catch_backtrace()))
+
+    finally
+        if out_buf != C_NULL
+            Libc.free(out_buf)
+        end
+    end
+
+    return Csize_t(nbytes_out)
+end
+
+# Filter registration
+
+# All information for the filter
+
+struct BitshuffleFilter <: Filter
+    major::Cuint
+    minor::Cuint
+    typesize::Cuint
+    blocksize::Cuint
+    compression::Cuint
+    comp_level::Cuint #Zstd only
+end
+
+"""
+    BitshuffleFilter(blocksize=0,compression="",comp_level=0)
+
+The Bitshuffle filter can optionally include compression "lz4" or "zstd". For "zstd"
+comp_level can be provided. This is ignored for "lz4" compression. If `blocksize`
+is zero the default bitshuffle blocksize is used.
+"""
+function BitshuffleFilter(;blocksize = 0, compressor="",comp_level=0)
```

Comment:
Thanks for the formatting. I wonder if using symbols `:lz4, :zstd, or :none` might be cleaner. It also might be good to explicitly check for `:none` so that typos do not fail silently. If someone typed "lz4d" or `:cstd` as a compressor, a useful error should be emitted.

Reply:
OK makes sense, will update to use symbols.

Reply:
Changed to symbols in latest commit. `:none` is included in the list of allowed symbols that are checked and the error message should be sufficiently informative I think.